src/smt/theory_lra.cpp

```diff
@@ -2215,10 +2215,13 @@ class theory_lra::imp {
         m_bp.init();
         lp().propagate_bounds_for_touched_rows(m_bp);
 
-        if (!m.inc()) 
+        if (!m.inc())
             return;
 
-        if (is_infeasible()) {
+        if (lp().get_status() == lp::lp_status::CANCELLED) {
```

Comment:
this looks fragile. Is there some common place this can be handled? 
TODO: establish when m.inc() is true but lp_status is CANCELLED.

Reply:
maybe add to return status of propagate_bounds_for_touched_rows()?